src/app/manage-withdrawals/manage-withdrawals.component.ts

```diff
@@ -24,7 +25,10 @@ export class ManageWithdrawalsComponent implements OnInit {
         SpinnerUtil.showSpinner();
         return this.withdrawCooperativeService.getApprovedWithdrawals().subscribe((res: any) => {
             SpinnerUtil.hideSpinner();
-            this.withdrawals = res.withdraws;
+            this.withdrawals = res.withdraws.map(x => {
+                x.amount = centsToBaseCurrencyUnit(x.amount);
+                return x;
+            })
```

Comment:
I don't know how urgent this pull request is, but I've got some points that we should discuss, and as "we" I mean the entire frontend team. I'll just mention them here to have the trace.

We should differentiate data models received from a backend API and ones that we use as a presentation logic. In other words, in this example, if an API returned data model that satisfies the `Withdraw` interface, we should not mutate this data and use it on the view side (in `*.component.html` files).
The better approach is to make another component-scoped interface called `WithdrawView` which will contain all necessary data required for presentation, including modified fields and newly calculated values.
Furthermore, it would better to avoid modifying data if it's only modified in favor of presentation logic. In these cases, using Angular pipes would be a more convenient way for this conversion from cents to base currency unit.

Tell me what you think :)

Reply:
I agree 100% with this. It's something I wanted to do for a while but never found the time to do it. However, merge this file now because it fixes two major bugs.